cookbook/controller/service.rst

```diff
@@ -232,6 +234,55 @@ inject *only* the exact service(s) that you need directly into the controller.
    are valid, exactly how you want to organize your reusable code is up to
    you.
 
+Base Controller Methods and their Service Replacements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This table explains how to replace the convenience methods of the base
+controller.
+
++-----------------------------+----------------------+-----------------------------------------------------------------+
+| Method                      | Service              |  PHP Code                                                       |
++=============================+======================+=================================================================+
+| ``createForm``              | ``form.factory``     | ``$formFactory->create($type, $data, $options)``                |
++-----------------------------+----------------------+-----------------------------------------------------------------+
+| ``createFormBuilder``       | ``form.factory``     | ``$formFactory->createBuilder('form', $data, $options)``        |
++-----------------------------+----------------------+-----------------------------------------------------------------+
+| ``createNotFoundException`` | \-                   | ``throw new NotFoundHttpException($message, $previous);``       |
++-----------------------------+----------------------+-----------------------------------------------------------------+
+| ``forward``                 | ``http_kernel``      | ``$httpKernel->forward($controller, $path, $query)``            |
++-----------------------------+----------------------+-----------------------------------------------------------------+
+| ``generateUrl``             | ``router``           | ``$router->generate($route, $params, $absolute)``               |
++-----------------------------+----------------------+-----------------------------------------------------------------+
+| ``getDoctrine``             | ``doctrine``         | *Simply inject doctrine instead of fetching from the container* |
++-----------------------------+----------------------+-----------------------------------------------------------------+
+| ``getUser``                 | ``security.context`` | $user = null;                                                   |
+|                             |                      | $token = $securityContext->getToken();                          |
+|                             |                      | if (null !== $token && is_object($token->getUser())) {          |
+|                             |                      | $user = $token->getUser();                                      |
+|                             |                      | }                                                               |
```

Comment:
this renders quite badly (all code lines floating) but i found no way how to include multiline source code in a table cell... ideas? btw, the indention is missing on purpose as otherwise we get a compile error.